app/utils/asyncio_utils.py

```diff
@@ -30,6 +32,22 @@ def __init__(self, *, max_concurrency: int = 0):
         else:
             self._semaphore = None
 
+    T = TypeVar("T")
+
+    @classmethod
+    async def run(
+        cls, *args: Coroutine[Any, Any, T], max_concurrency: int
+    ) -> list[asyncio.Future[T]]:
+        """
+
+        @param args: coroutines
+        @param max_concurrency: the number of concurrent tasks running
+        @return: dispatched tasks
+        """
+        async with SemaphoreTaskGroup(max_concurrency=max_concurrency) as tg:
+            dispatched = [tg.create_task(coro) for coro in args]
+        return dispatched
+
```

Comment:
To reduce the amount of code for asynchronous execution using `SemaphoreTaskGroup` and to simplify it, added `run` function as a `@classmethod`.
The usage is like following.
```Python
tasks = await SemaphoreTaskGroup.run(
    coroA(),
    coroB(),
    coroC(),
    coroD(),
    max_concurrency=3
)
results = [task.result() for task in tasks]
```